src/cfi.rs

```diff
@@ -63,6 +63,78 @@ impl<'input, Endian> DebugFrame<'input, Endian>
         let (_, entry) = try!(CommonInformationEntry::parse(input));
         Ok(entry)
     }
+
+    /// Find the frame unwind information for the given address.
+    ///
+    /// If found, the unwind information is returned along with the reset
+    /// context in the form `Ok((unwind_info, context))`. If not found,
+    /// `Err(gimli::Error::NoUnwindInfoForAddress)` is returned. If parsing or
+    /// CFI evaluation fails, the error is returned.
+    ///
+    /// ```
+    /// use gimli::{DebugFrame, NativeEndian, UninitializedUnwindContext};
+    ///
+    /// # fn foo() -> gimli::Result<()> {
+    /// # let read_debug_frame_section = || unimplemented!();
+    /// // Get the `.debug_frame` section from the object file.
+    /// let debug_frame = DebugFrame::<NativeEndian>::new(read_debug_frame_section());
+    ///
+    /// # let get_frame_pc = || unimplemented!();
+    /// // Get the address of the PC for a frame you'd like to unwind.
+    /// let address = get_frame_pc();
+    ///
+    /// // This context is reusable, which cuts down on heap allocations.
+    /// let ctx = UninitializedUnwindContext::new();
+    ///
+    /// let (unwind_info, ctx) = try!(debug_frame.unwind_info_for_address(ctx, address));
+    /// # let do_stuff_with = |_| unimplemented!();
+    /// do_stuff_with(unwind_info);
+    /// # let _ = ctx;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn unwind_info_for_address
+        (&self,
+         ctx: UninitializedUnwindContext<'input, Endian>,
+         address: u64)
+         -> Result<(UnwindTableRow<'input, Endian>, UninitializedUnwindContext<'input, Endian>)> {
+        let mut target_fde = None;
+
+        let mut entries = self.entries();
+        while let Some(entry) = try!(entries.next()) {
+            match entry {
+                CieOrFde::Cie(_) => continue,
+                CieOrFde::Fde(partial) => {
+                    let fde = try!(partial.parse(|offset| self.cie_from_offset(offset)));
```

Comment:
Unfortunately, you need to parse the CIE to figure out the FDE's range of addresses :(

We could build up a map of offset -> CIE while iterating entries, because it seems likely that the CIE will come before the FDE, but I'll leave that to when we have benches and can properly investigate.